lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -3029,11 +3037,22 @@ private function unregisterDocument($document)
      */
     public function registerDocument($document, $id)
     {
-        $oid = spl_object_hash($document);
+    	$oid = spl_object_hash($document);
         $this->documentIds[$oid] = $id;
         $this->identityMap[$id] = $document;
-        $this->setDocumentState($oid, self::STATE_MANAGED);
-
+        
+        // frozen nodes need another state so they are managed but not included for updates
+        $nodeType = '';
+        try {
+        	$node = $this->session->getNode($id);
+        	// In some of the unit tests $node will be null because of how they are mocked
+        	$nodeType = $node ? $node->getPrimaryNodeType()->getName() : '';
+        } catch (PathNotFoundException $e) {
+        	// Can happen if the document is new and not yet persisted
+        }
+        
+        $this->setDocumentState($oid, $nodeType=="nt:frozenNode" ? self::STATE_FROZEN : self::STATE_MANAGED);
```

Comment:
Better to not inline this..

````php
$documentState = $nodeType == 'nt:frozenNode' ? self::STATE_FROZEN : self::STATE_MANAGED;
$this->setDocumentState($oid, $documentState);
````

But then I think I might move all of this into a separate method .. do we need this logic in other places?